rust/src/dataset/scanner.rs

```diff
@@ -388,12 +407,15 @@ impl Scanner {
         filter: Arc<dyn PhysicalExpr>,
         input: Arc<dyn ExecutionPlan>,
         drop_row_id: bool,
+        ann_schema: Option<Arc<Schema>>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let filter_node = Arc::new(FilterExec::try_new(filter, input)?);
+        let output_schema = self.scanner_output_schema()?;
         Ok(Arc::new(LocalTakeExec::new(
             filter_node,
             self.dataset.clone(),
-            Arc::new(self.projections.clone()),
+            output_schema,
```

Comment:
this is leaking the abstraction of ann to take?

what is in output_schema nad `ann_schema` tho? can we just use one?

Reply:
output_schema includes user supplied projections. ann_schema includes the vector/score columns.